formats/dygraphs/SRF_Dygraphs.php

```diff
@@ -90,27 +90,29 @@ protected function getResultData( SMWQueryResult $result, $outputMode ) {
 				while ( ( $dataValue = $field->getNextDataValue() ) !== false ) { // Data values
 
 					// Jump the column (indicated by continue) because we don't want the data source being part of the annotation array
-					if ( $dataValue->getDataItem()->getDIType(
-						) == SMWDataItem::TYPE_WIKIPAGE && $this->params['datasource'] === 'raw' && !$dataSource ) {
+					$dataItem = $dataValue->getDataItem();
+
+					if ( $dataItem->getDIType() === SMWDataItem::TYPE_WIKIPAGE ) {
+						$title = $dataItem->getTitle();
+					}
```

Comment:
This is a bit scary... $title might end up not being defined or still be from the previous iteration. Probably need to extract some functions to make this code not scary.